### PR TITLE
Add new template for rabbit-mq trigger

### PR DESCRIPTION
**Pull request by Flanker32**
Add RabbitMQ Trigger for `azure-functions:add`, for issue https://github.com/microsoft/azure-maven-plugins/issues/1197

``` java
package com.functions;

import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.rabbitmq.annotation.RabbitMQTrigger;

/**
 * Azure Functions with RabbitMQ Trigger.
 *
 * This function will be invoked when a new message is received in RabbitMQ
 * Please add com.microsoft.azure.functions:azure-functions-java-library-rabbitmq to your project dependencies
 * You may get the example and latest version in https://search.maven.org/artifact/com.microsoft.azure.functions/azure-functions-java-library-rabbitmq
 * 
 */
public class RabbitMQTriggerJava {
    @FunctionName("RabbitMQTriggerJava")
    public void run(
            @RabbitMQTrigger(
                    connectionStringSetting = "RabbitMQSetting",
                    queueName = "test"
            ) String input,
            final ExecutionContext context) {
        context.getLogger().info("Java RabbitMQ trigger processed a request." + input);
    }
}
```

### CODE REVIEW

File: azure-tools-common/src/main/resources/resources.json [hunk 1]
```diff
@@ -58,6 +58,12 @@
     "cosmosDBIn_leaseCollectionName_label": "Collection name for leases",
     "cosmosDBIn_leaseCollectionName_help": "Name of the collection to store the leases.",
     "cosmosDBIn_createIfNotExists_label": "Create lease collection if it does not exist",
-    "cosmosDBIn_createIfNotExists_help": "Checks for existence and automatically creates the leases collection."
+    "cosmosDBIn_createIfNotExists_help": "Checks for existence and automatically creates the leases collection.",
+    "RabbitMQTrigger_description" : "A function that will be run whenever a message is added to a specified RabbitMQ queue",
```
Comment (andxu) [inline azure-tools-common/src/main/resources/resources.json:62]: whenever -> when

Comment (Flanker32) [reply azure-tools-common/src/main/resources/resources.json:62]: Refers https://github.com/Azure/azure-functions-templates/pull/995/files#diff-80abd3bf444ae5625706b62384f92b234615dc3a73d5aae76ffd11595ad7e099R937, keep consistent with Function template

**Comment by pragnagopa** [comment]
@yojagad  - Please review

@Flanker32  - Please let us know if there is a way test this on our end before merging. Thanks!

**Review by yojagad — APPROVED** [review]

**Comment by Flanker32** [comment]
> @yojagad - Please review
> 
> @Flanker32 - Please let us know if there is a way test this on our end before merging. Thanks!

@pragnagopa functions maven plugin 1.11.0-SNAPSHOT has been released to sonatype, you may try it with the following configuration in `pom.xml`
```
    <pluginRepositories>
        <pluginRepository>
            <id>sonatype</id>
            <name>sonatype</name>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
            <releases>
                <enabled>false</enabled>
            </releases>
        </pluginRepository>
     </pluginRepositories>
```

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**